src/compiler/crystal/command.cr

```diff
@@ -449,14 +449,17 @@ class Crystal::Command
         opts.on("--release", "Compile in release mode (-O3 --single-module)") do
           compiler.release!
         end
-        opts.on("-O3", "LLVM optimization level 3") do
-          compiler.optimization_mode = Compiler::OptimizationMode::O3
-        end
-        opts.on("-O2", "LLVM optimization level 2") do
-          compiler.optimization_mode = Compiler::OptimizationMode::O2
-        end
-        opts.on("-O1", "LLVM optimization level 1") do
-          compiler.optimization_mode = Compiler::OptimizationMode::O1
+        opts.on("-O LEVEL", "LLVM optimization level") do |level|
+          case level.to_i
+          when 3
+            compiler.optimization_mode = Compiler::OptimizationMode::O3
+          when 2
+            compiler.optimization_mode = Compiler::OptimizationMode::O2
+          when 1
+            compiler.optimization_mode = Compiler::OptimizationMode::O1
+          else
+            raise Error.new("Unknown optimization level: #{level}")
+          end
```

Comment:
What about just doing like:

```suggestion
        opts.on("-O LEVEL", "LLVM optimization level") do |level|
          compiler.optimization_mode = Compiler::OptimizationMode.from_value level.to_i
```

Reply:
You don't really need to manually raise an error for unknown levels. The non nilable from value method already handles that for you.

Reply:
but it need to raise special error, to get properly catched.